utils/supabase/queries.ts

```diff
@@ -37,3 +43,24 @@ export const getUserDetails = cache(async (supabase: SupabaseClient) => {
     .single();
   return userDetails;
 });
+
+export const getUserSubscriptions = cache(async (supabase: SupabaseClient, user: User | null) => {
+  if (!user) {
+    return [];
+  }
+  const { data: subscriptions, error } = await supabase
+    .from('subscriptions')
+    .select('*, prices(*, products(*))')
+    .eq('user_id', user.id)
```

Comment:
This is not necessary because subscriptions table in Supabase has enabled Row-Level Security (RLS) policies, meaning that, users can only access their own data.